frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/OverrideForkPointBackgroundable.java

```diff
@@ -0,0 +1,100 @@
+package com.virtuslab.gitmachete.frontend.actions.backgroundables;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vfs.VirtualFile;
+import git4idea.config.GitConfigUtil;
+import git4idea.repo.GitRepository;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.backend.api.ICommitOfManagedBranch;
+import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot;
+import com.virtuslab.gitmachete.backend.api.INonRootManagedBranchSnapshot;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.OverrideForkPointDialog;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@CustomLog
+public class OverrideForkPointBackgroundable extends Task.Backgroundable {
+
+  private final GitRepository gitRepository;
+
+  private final INonRootManagedBranchSnapshot nonRootBranch;
+
+  @Nullable
+  private final ICommitOfManagedBranch selectedCommit;
+
+  private final BaseEnhancedGraphTable graphTable;
+
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  public OverrideForkPointBackgroundable(Project project, String title, GitRepository gitRepository,
+      INonRootManagedBranchSnapshot nonRootBranch, BaseEnhancedGraphTable graphTable) {
+    super(project, title);
+    this.gitRepository = gitRepository;
+    this.nonRootBranch = nonRootBranch;
+    this.graphTable = graphTable;
+    selectedCommit = new OverrideForkPointDialog(project, nonRootBranch).showAndGetSelectedCommit();
```

Comment:
Hmmm I wouldn't execute such action in backgroundable's c'tor 🤔 move dialog creation back to the call site

Reply:
good point. moved